scripts/upgrade/generate-upgrade-config.ts

```diff
@@ -140,6 +156,20 @@ async function generateUpgradeConfig(
       }
     }
 
+    const metaTxAllowlistAdd = newContracts
+      .filter((contract) => contract.endsWith("Facet") && !contract.includes("/test/") && !contract.includes("Mock"))
+      .flatMap((contractName) => {
+        const details = targetBytecodes[contractName]?.functionSignatures || [];
+        return details.map((detail) => ({ facetName: contractName, functionName: detail.name, hash: detail.hash }));
+      });
+
```

Comment:
You also need to take into account the changes in `changedContracts`, i.e. find the selectors that existed in old facets, but not anymore, and find the selectors that were introduced in the target version.